src/AFQMC/AFQMCFactory.cpp

```diff
@@ -198,28 +199,32 @@ bool AFQMCFactory::parse(xmlNodePtr cur)
 
 bool AFQMCFactory::execute(xmlNodePtr cur)
 {
-  if (cur == NULL)
+  if (cur == nullptr)
     return false;
 
   int groupid = 0; //myComm->getGroupID();
-  char fileroot[256];
+  std::array<char, 256> fileroot;
 
   bool no_gtag = (qmc_common.mpi_groups == 1);
 
   xmlNodePtr curRoot = cur;
   cur                = curRoot->children;
-  while (cur != NULL)
+  while (cur != nullptr)
   {
     std::string cname((const char*)(cur->name));
     if (cname == "execute")
     {
+      int length{0};
       if (no_gtag) //qnproc_g == nproc)
-        sprintf(fileroot, "%s.s%03d", project_title.c_str(), m_series);
+        length = snprintf(fileroot.data(), fileroot.size(), "%s.s%03d", project_title.c_str(), m_series);
       else
-        sprintf(fileroot, "%s.g%03d.s%03d", project_title.c_str(), groupid, m_series);
+        length = snprintf(fileroot.data(), fileroot.size(), "%s.g%03d.s%03d", project_title.c_str(), groupid, m_series);
+
+      if (s_size < 0)
```

Comment:
Should this be 'length' instead of 's_size'?

Reply:
:facepalm: Should be fixed now.